apps/studio/components/interfaces/Auth/ThirdPartyAuthForm/CreateAuth0Dialog.tsx

```diff
@@ -0,0 +1,182 @@
+
+const FormSchema = z.object({
+  enabled: z.boolean(),
+  auth0DomainName: z
+    .string()
+    .trim()
+    .min(1)
+    .regex(/^[A-Za-z0-9-.]+$/, 'The project ID contains invalid characters.'), // Only allow alphanumeric characters and hyphens.
+})
+
+export const CreateAuth0IntegrationDialog = ({
+  visible,
+  onClose,
+  onDelete,
+}: CreateAuth0IntegrationProps) => {
+  // TODO: Remove this if this Dialog is only used for creating.
+  const isCreating = true
+
+  const { ref: projectRef } = useParams()
+  const { mutate: createAuthIntegration, isLoading } = useCreateThirdPartyAuthIntegrationMutation({
+    onSuccess: () => {
+      toast.success(`Successfully created a new Auth0 Auth integration.`)
+      onClose()
+    },
+  })
+
+  const form = useForm<z.infer<typeof FormSchema>>({
+    resolver: zodResolver(FormSchema),
+    defaultValues: {
+      enabled: true,
+      auth0DomainName: '',
+    },
+  })
+
+  useEffect(() => {
+    if (visible) {
+      form.reset({
+        enabled: true,
+        auth0DomainName: '',
+      })
+
+      // the form input doesn't exist when the form is reset
+      setTimeout(() => {
+        form.setFocus('auth0DomainName')
+      }, 25)
+    }
+  }, [visible])
+
+  const onSubmit: SubmitHandler<z.infer<typeof FormSchema>> = async (values) => {
+    createAuthIntegration({
+      projectRef: projectRef!,
+      oidcIssuerUrl: `https://${values.auth0DomainName}.auth0.com`,
+    })
+  }
+
+  return (
+    <Dialog open={visible} onOpenChange={() => onClose()}>
+      <DialogContent>
+        <DialogHeader>
+          <DialogTitle className="truncate">
+            {isCreating ? `Add new Auth0 connection` : `Update existing Auth0 connection`}
+          </DialogTitle>
+        </DialogHeader>
+        <Separator />
+        <DialogSection>
+          <Form_Shadcn_ {...form}>
+            <form id={FORM_ID} onSubmit={form.handleSubmit(onSubmit)} className="space-y-4">
+              {/* Enabled flag can't be changed for now because there's no update API call for integrations */}
+              {/* <FormField_Shadcn_
+              key="enabled"
+              control={form.control}
+              name="enabled"
+              render={({ field }) => (
+                <FormItemLayout
+                  className="px-8"
+                  label={`Enable Auth0 Auth Connection`}
+                  layout="flex"
+                >
+                  <FormControl_Shadcn_>
+                    <Switch
+                      checked={field.value}
+                      onCheckedChange={field.onChange}
+                      disabled={field.disabled}
+                    />
+                  </FormControl_Shadcn_>
+                </FormItemLayout>
+              )}
+            />
+            <Separator /> */}
+              <p className="text-sm text-foreground-light">
+                This will enable a JWT token from Auth0 project to access data from this Supabase
```

Comment:
"Enable a JWT token" sounds a little weird - might need more context as to what is happening here to decide on a better language

```suggestion
                This will enable a JWT token from your Auth0 project to access data from this Supabase
```

Reply:
Fixed.